### ISSUE TITLE
Name shadowing warning in generated code

### ISSUE THREAD

**Issue by merijn** [yesodweb/persistent#1493]
Continuation of #1406? I'm using persistent 2.14.5.0 and I'm still getting name-shadowing errors in generated code. The culprits appear to be `keyFromRecordM` and `fieldLens`. Example splices generated where GHC complains that `globalVarsName` is shadowed:

```haskell
    Database.Persist.Class.PersistEntity.keyFromRecordM
        = Just
            (\ GlobalVars {globalVarsName = globalVarsName}
               -> GlobalVarsKey globalVarsName)
```

Above `globalVarsName` is used as variable name for the field with the corresponding name.

```haskell
      Database.Persist.Class.PersistEntity.fieldLens GlobalVarsValue
        = (Database.Persist.TH.lensPTH
             ((\ GlobalVars {globalVarsValue = x} -> x)
                . Database.Persist.Class.PersistEntity.entityVal))
            (\ (Database.Persist.Class.PersistEntity.Entity key_a1mqV
                                                            value_a1mqW)
               x_a1mqX
               -> (Database.Persist.Class.PersistEntity.Entity key_a1mqV)
                    (case value_a1mqW of {
                       GlobalVars {globalVarsName = globalVarsName_a1mr0}
                         -> GlobalVars
                              {globalVarsName = globalVarsName, globalVarsValue = x_a1mqX} }))
```

This splice is even more confusing, as it seems to use `globalVarsName` in the creation of a `GlobalVars` record, but it doesn't even seem to be in scope?

**Comment by hw202207** [comment]
I think either #1496 or #1494 would fix this problem.

**Comment by parsonsmatt** [comment]
Closed by #1496
